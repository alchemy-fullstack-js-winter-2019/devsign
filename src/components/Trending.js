import React from 'react';

export default function Trending() {
  return (
    <aside>
      <h3>Trending</h3> 
      <ul>
        <li>#Trending 1</li>
        <li>#Trending 2</li>
        <li>#Trending 3</li>
        <li>#Trending 4</li>
        <li>#Trending 5</li>
      </ul>
    </aside>
  );
}

