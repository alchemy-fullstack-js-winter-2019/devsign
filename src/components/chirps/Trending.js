import React from 'react';
import styled from 'styled-components';

const Li = styled.li`
  list-style-type: none;
  margin-bottom: 5px;
`;

const Ul = styled.ul`
  margin-bottom: 10px;
  padding: 0;
`;

const Aside = styled.aside`
  border: 3px solid blue;
  margin-top: 10px;
  text-align: center;
  h3 {
    font-weight: 200;
  }
`;

export default function Trending() {
  return (
    <Aside>
      <h3>Trending</h3> 
      <Ul>
        <Li>#Bro</Li>
        <Li>#Blessed</Li>
        <Li>#NoFilter</Li>
        <Li>#HumbleBrag</Li>
        <Li>#JustSaying</Li>
        <Li>#NationalWineDay</Li>
        <Li>#RipCity</Li>
        <Li>#WCW</Li>
        <Li>#MCM</Li>
      </Ul>
    </Aside>
  );
}

