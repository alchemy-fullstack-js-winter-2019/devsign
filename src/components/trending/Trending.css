.Trending {
  padding: 40px 50px;
  background-color: #F1BF98 ;
  text-align: center;
  margin-right: auto;
}
img {
  width: 100px;
  height: 100px;
}
@media only screen and (max-width: 600px) {
  .Trending {
    background-color: lightgray;
    color: white;
  }
}
