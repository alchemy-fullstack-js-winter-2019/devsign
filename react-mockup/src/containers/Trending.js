import { connect } from 'react-redux';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { getQuotes } from '../selectors/trending';
import TopPosts from '../components/user/TopPosts';
import { fetchQuotes } from '../actions/trending';

class Trending extends PureComponent {
  static propTypes = {
    quotes: PropTypes.array.isRequired,
    fetch: PropTypes.func.isRequired
  }

  componentDidMount() {
    this.props.fetch();
  }

  render() {
    return (
      <TopPosts quotes={this.props.quotes} />
    );
  }
}

const mapStateToProps = state => ({
  quotes: getQuotes(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchQuotes());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Trending);


