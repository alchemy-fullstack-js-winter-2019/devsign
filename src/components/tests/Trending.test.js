import React from 'react';
import renderer from 'react-test-renderer';
import Trending from '../chirps/Trending';

describe('Trending', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Trending />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
