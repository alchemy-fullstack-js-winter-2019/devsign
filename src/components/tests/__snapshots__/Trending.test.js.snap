// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending matches a snapshot 1`] = `
<aside
  className="sc-htpNat bdCFJT"
>
  <h3>
    Trending
  </h3>
  <ul
    className="sc-bwzfXH qSXGG"
  >
    <li
      className="sc-bdVaJa BwNPC"
    >
      #Bro
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #Blessed
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #NoFilter
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #HumbleBrag
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #JustSaying
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #NationalWineDay
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #RipCity
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #WCW
    </li>
    <li
      className="sc-bdVaJa BwNPC"
    >
      #MCM
    </li>
  </ul>
</aside>
`;
